hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -96,7 +101,8 @@ public void initialize() {
     // this operation returns the value as Data, except when it's retrieved from Near Cache with in-memory-format OBJECT
     @Override
     @SuppressWarnings("unchecked")
-    protected V getInternal(Data key) {
+    protected V getInternal(Object key) {
+        key = serializeKeys ? toDataWithStrategy(key) : key;
```

Comment:
Seems we should always convert key to `Data` if it is not found in near cache. Otherwise there can be redundant `Data` conversions for various reasons.